include/aikido/util/detail/yaml_extension.hpp

```diff
@@ -0,0 +1,246 @@
+
+      for (Index c = 0; c < matrix.cols(); ++c)
+        row.push_back(matrix(r, c));
+
+      node.push_back(row);
+    }
+
+    return node;
+  }
+};
+
+} // namespace (anonymous)
+
+namespace YAML {
+
+//==============================================================================
+// Specialization for Eigen::Matrix<...>. This enables to use YAML::Node
+// with std::unordered_map as `Node(Eigen::Matrix3d::Identity())` and
+// `node.as<Eigen::Matrix3d>()`.
+template <typename _Scalar, int _Dim, int _Mode, int _Options>
+struct convert<Eigen::Matrix<_Scalar, _Dim, _Mode, _Options>>
+{
+  using MatrixType = Eigen::Matrix<_Scalar, _Dim, _Mode, _Options>;
+
+  static Node encode(const MatrixType& matrix)
+  {
+    return encode_impl<MatrixType, MatrixType::IsVectorAtCompileTime>::encode(
+        matrix);
+  }
+
+  static bool decode(
+      const YAML::Node& node,
+      Eigen::Matrix<_Scalar, _Dim, _Mode, _Options>& matrix)
+  {
+    if (node.Tag() == "Vector" || node.Tag() == "!Vector"
+        || node.Tag() == "Matrix"
+        || node.Tag() == "!Matrix")
+    {
+      ::decode(node, matrix);
+      return true;
+    }
+    else
+    {
+      return false;
+    }
+  }
+};
+
+//==============================================================================
+// Specialization for Eigen::Isometry<...>. This enables to use YAML::Node
+// with std::unordered_map as `Node(Eigen::Isometry3d::Identity())` and
+// `node.as<Eigen::Isometry3d>()`.
+template <typename _Scalar, int _Dim, int _Mode, int _Options>
+struct convert<Eigen::Transform<_Scalar, _Dim, _Mode, _Options>>
+{
+  using TransformType = Eigen::Transform<_Scalar, _Dim, _Mode, _Options>;
+  using MatrixType = typename TransformType::MatrixType;
+
+  static Node encode(const TransformType& transform)
+  {
+    return convert<MatrixType>::encode(transform.matrix());
+  }
+
+  static bool decode(const Node& node, TransformType& transform)
+  {
+    return convert<MatrixType>::decode(node, transform.matrix());
+  }
+};
+
+//==============================================================================
+// Specialization for std::unordered_map<...>. This enables to use YAML::Node
+// with std::unordered_map as `Node(std::unorderd_map<...>())` and
+// `node.as<std::unordered_map<...>>()`.
+template <typename _Key,
+          typename _Tp,
+          typename _Hash,
+          typename _Pred,
+          typename _Alloc>
+struct convert<std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>>
+{
+  using UnorderedMap = std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>;
+
+  static Node encode(const UnorderedMap& map)
+  {
+    Node node(NodeType::Map);
+
+    for (const auto& it : map)
+      node.force_insert(it.first, it.second);
+
+    return node;
+  }
+
+  static bool decode(const Node& node, UnorderedMap& map)
+  {
+    if (!node.IsMap())
+      return false;
+
+    map.clear();
+    for (const auto& it : node)
+      map[it.first.as<_Key>()] = it.second.as<_Tp>();
```

Comment:
What happens if `as<>()` fails? If we want to return `false` on failure, then we need to handle that case explicitly.